specs/_features/ePBS/p2p-interface.md

```diff
@@ -0,0 +1,119 @@
+
+###### `beacon_block`
+
+The *type* of the payload of this topic changes to the (modified) `SignedBeaconBlockWithBid` found in ePBS spec. Specifically, this type changes with the replacement of `SignedBuilderBid` to the inner `BeaconBlockBody`.
+
+In addition to the gossip validations for this topic from prior specifications, the following validations MUST pass before forwarding the `signed_beacon_block` on the network. Alias `block = signed_beacon_block.message`, `signed_bid = block.body.signed_builder_bid`, `bid = signed_bid.message`, `header = bid.header`.
+
+
+New validation:
+
+- _[REJECT]_ The block's execution header timestamp is correct with respect to the slot -- i.e. `header.timestamp == compute_timestamp_at_slot(state, block.slot)`.
+- _[REJECT]_ The bid pubkey is a valid builder in state.
+- _[REJECT]_ The builder has sufficient balances to pay for the bid.
+- _[REJECT]_ The builder signature, `signed_bid.signature`, is valid with respect to the `bid.pubkey`.
+
+###### `execution_payload`
+
+This topic is used to propagate execution payload.
+
+The following validations MUST pass before forwarding the `execution_payload` on the network, assuming the alias `payload = execution_payload`:
+
+- _[IGNORE]_ The payload aligns with header that received in block.
+
+###### `execution_attestation`
+
+This topic is used to propagate signed execution attestation.
+
+The following validations MUST pass before forwarding the `execution_attestation` on the network, assuming the alias `data = execution_attestation.data`:
+
+- _[IGNORE]_ `data.slot` is within the last `ATTESTATION_PROPAGATION_SLOT_RANGE` slots (within a MAXIMUM_GOSSIP_CLOCK_DISPARITY allowance) -- i.e. `data.slot + ATTESTATION_PROPAGATION_SLOT_RANGE >= current_slot >= data.slot` (a client MAY queue future attestations for processing at the appropriate slot).
+- _[REJECT]_ The validator index is within the execution committee in `get_execution_committee(state, data.slot)`
+- _[IGNORE]_ The execution payload being voted for (`data.execution_hash`) has been seen (via both gossip and non-gossip sources) (a client MAY queue execution attestations for processing once payload is retrieved).
+- _[REJECT]_ The signature of `execution_attestation` is valid.
+
+###### `builder_bid`
+
+This topic is used to propagate signed builder bids.
+
+The following validations MUST pass before forwarding the `signed_builder_bid` on the network, assuming the alias `bid = signed_builder_bid.message`:
+
+- _[REJECT]_ The signed builder bid pubkey, `bid.pubkey`, exists in state.
+- _[IGNORE]_ The signed builder bid value, `bid.value`, is less than builder's balance in state.
+- _[IGNORE]_ The signed builder header timestamp is correct with respect to next slot -- i.e. `header.timestamp == compute_timestamp_at_slot(state, current_slot + 1)`.
+- _[IGNORE]_ The signed builder header parent block has matches one of the chain tip in the fork choice store. Builder may submit multiple bids with respect to forks.
```

Comment:
We have to allow building on any viable block in the DAG or it can cause liveness issues. The builder's beacon node may require to fork an existing tip

Reply:
Yeah. That's fine. The comment is saying the builder's bid has to build on a parent block hash that is seen. Perhaps I should rephrase it differently

Reply:
```suggestion
- _[IGNORE]_ The signed builder header parent block matches one of the chain tip(s) in the fork choice store. Builder may submit multiple bids corresponding to various forks.
```